src/apps/components/AppFrame/useAppActions.ts

```diff
@@ -29,16 +31,36 @@ export const useAppActions = (
       case "redirect": {
         const { to, newContext, actionId } = action.payload;
 
-        if (newContext) {
-          window.open(to);
-        } else if (to.startsWith("/")) {
-          navigate(to);
-          closeApp();
-        } else {
-          window.location.href = to;
+        let success = true;
```

Comment:
I dont feel comfortable looking a longer than 3 loc block of code under switch-case, named fn maybe?